sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala

```diff
@@ -59,19 +57,17 @@ case class BroadcastHashJoin(
   override def requiredChildDistribution: Seq[Distribution] =
     UnspecifiedDistribution :: UnspecifiedDistribution :: Nil
 
-  @transient
-  private val broadcastFuture = future {
+  private lazy val broadcastedRelation = {
```

Comment:
I think so, will re-purpose this PR to fix the thead-safy issue about UnsafeProjection.